utils/utils.go

```diff
@@ -325,28 +325,55 @@ func CustomizeNetworkPolicy(networkPolicy *networkingv1.NetworkPolicy, ba common
 		networkPolicy.Spec.Ingress[0].From = append(networkPolicy.Spec.Ingress[0].From, networkingv1.NetworkPolicyPeer{})
```

Comment:
By creating a new Ingress and setting it to `networkPolicy.Spec.Ingress`, the `len` checks above and below can be removed. The Ingress rule would always be created from scratch based on the CR values.